### PR TITLE
Add DNS Provider for inwx

### PR DESCRIPTION
**Pull request by smueller18**
Add support for provider INWX

Livetest was successful obtaining the following results (replaced actual domain with example.org):
```console
$ INWX_USERNAME=$USERNAME INWX_PASSWORD=$PASSWORD INWX_DOMAIN=$DOMAIN go test -v -cover
=== RUN   TestNewDNSProvider
=== RUN   TestNewDNSProvider/success
2018/10/25 18:20:55 [INFO] inwx: sandbox mode is enabled
=== RUN   TestNewDNSProvider/missing_credentials
=== RUN   TestNewDNSProvider/missing_username
=== RUN   TestNewDNSProvider/missing_password
--- PASS: TestNewDNSProvider (0.00s)
    --- PASS: TestNewDNSProvider/success (0.00s)
    --- PASS: TestNewDNSProvider/missing_credentials (0.00s)
    --- PASS: TestNewDNSProvider/missing_username (0.00s)
    --- PASS: TestNewDNSProvider/missing_password (0.00s)
=== RUN   TestNewDNSProviderConfig
=== RUN   TestNewDNSProviderConfig/success
=== RUN   TestNewDNSProviderConfig/missing_credentials
--- PASS: TestNewDNSProviderConfig (0.00s)
    --- PASS: TestNewDNSProviderConfig/success (0.00s)
    --- PASS: TestNewDNSProviderConfig/missing_credentials (0.00s)
=== RUN   TestLivePresentAndCleanup
2018/10/25 18:20:55 [INFO] inwx: sandbox mode is enabled
--- PASS: TestLivePresentAndCleanup (1.48s)
PASS
coverage: 80.0% of statements
ok  	github.com/xenolf/lego/providers/dns/inwx	1.494s
```

```console
$ INWX_USERNAME=$USERNAME INWX_PASSWORD=$PASSWORD ./lego -d example.org -m mail@example.org --dns inwx --server=https://acme-staging-v02.api.letsencrypt.org/directory --dns-resolvers ns2.inwx.de --accept-tos run
2018/10/25 17:47:50 [INFO] [example.org] acme: Obtaining bundled SAN certificate
2018/10/25 17:47:51 [INFO] [example.org] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz/aWtFzgdxeB3zVRf2bJBZHj6MDUmESMOWoUBhrjib56o
2018/10/25 17:47:51 [INFO] [example.org] acme: Preparing to solve DNS-01
2018/10/25 17:47:51 [INFO] [example.org] acme: Trying to solve DNS-01
2018/10/25 17:47:51 [INFO] [example.org] Checking DNS record propagation using [ns2.inwx.de:53]
2018/10/25 17:47:51 [INFO] Wait [timeout: 1m0s, interval: 2s]
2018/10/25 17:47:57 [INFO] [example.org] The server validated our request
2018/10/25 17:47:58 [INFO] [example.org] acme: Validations succeeded; requesting certificates
2018/10/25 17:47:59 [INFO] [example.org] Server responded with a certificate.
```

### CODE REVIEW

File: providers/dns/inwx/inwx.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	}
+
+	return nil
+}
+
+// CleanUp removes the TXT record matching the specified parameters
+func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
+	fqdn, _, _ := acme.DNS01Record(domain, keyAuth)
+
+	// TODO check
```
[the remark this answers is not in the document]
Comment (smueller18) [reply providers/dns/inwx/inwx.go]: Damn, I forgot to remove it.

File: providers/dns/inwx/inwx_test.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+)
+
+var envTest = tester.NewEnvTest(
+	"INWX_USERNAME",
+	"INWX_PASSWORD").
+	WithDomain("INWX_DOMAIN")
+
+func init() {
+	os.Setenv("INWX_SANDBOX", "true")
+	os.Setenv("INWX_TTL", "3600")
```
[the remark this answers is not in the document]
Comment (smueller18) [reply providers/dns/inwx/inwx_test.go]: INWX offers a sandbox mode. In sandbox mode, the minimum TTL is 3600. I add a comment to clairify this.

File: providers/dns/inwx/inwx.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	values, err := env.Get("INWX_USERNAME", "INWX_PASSWORD")
+	if err != nil {
+		return nil, fmt.Errorf("inwx: %v", err)
+	}
+
+	config := NewDefaultConfig()
+	config.Username = values["INWX_USERNAME"]
+	config.Password = values["INWX_PASSWORD"]
+
+	if config.Sandbox {
```
[the remark this answers is not in the document]
Comment (smueller18) [reply providers/dns/inwx/inwx.go]: done

File: Gopkg.toml [hunk 4]
```diff
@@ -84,3 +84,7 @@
 [[constraint]]
   version = "0.11.1"
   name = "github.com/exoscale/egoscale"
+
+[[constraint]]
+  version = "v0.4.0"
+  name = "github.com/smueller18/goinwx"
```
[the remark this answers is not in the document]
Comment (smueller18) [reply Gopkg.toml]: My PR was not merged into steigr/goinwx so I used my own fork. Forgot to change the doc link.

File: providers/dns/inwx/inwx_test.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/xenolf/lego/platform/tester"
+)
+
+var envTest = tester.NewEnvTest(
+	"INWX_USERNAME",
+	"INWX_PASSWORD").
+	WithDomain("INWX_DOMAIN")
+
+func init() {
+	os.Setenv("INWX_SANDBOX", "true")
```
[the remark this answers is not in the document]
Comment (smueller18) [reply providers/dns/inwx/inwx_test.go]: You mean like this? https://github.com/xenolf/lego/pull/687/commits/08dc0cef63dc739dc2a8d3cfbd6b36d7a0423fb7

[the remark this answers is not in the document]
Comment (smueller18) [reply providers/dns/inwx/inwx_test.go]: This part is a bit tricky. The `config.Sandbox` and `config.TTL` values should only be changed if a live test is performed. These values are set by default in `NewDefaultConfig`. So the only way to influence this settings without influencing the regular test cases is to manually set the environments. In e82c2a5035a21789cbad5aaec151caba9df6ad2b I moved them to the live test section. If this is not desired, please tell me what I should change in more detail. Thanks

**Comment by ldez** [comment]
Could you do a test with the following command and give me the output:

```bash
./lego -m your@email.com -a -x http-01 -x tls-alpn-01 --dns inwx -s https://acme-staging-v02.api.letsencrypt.org/directory -d mydomain.com -d *.mydomain.com run
```

**Comment by smueller18** [comment]
Here is the output:
```
> INWX_PROPAGATION_TIMEOUT=240 INWX_USERNAME=$INWX_USERNAME INWX_PASSWORD=$INWX_PASSWORD ./lego -m your@email.com -a -x http-01 -x tls-alpn-01 --dns inwx -s https://acme-staging-v02.api.letsencrypt.org/directory -d mydomain.com -d "*.mydomain.com" run 
2018/11/04 20:02:37 No key found for account your@email.com. Generating a curve P384 EC key.
2018/11/04 20:02:37 Saved key to /home/sm/go/src/github.com/xenolf/lego/.lego/accounts/acme-staging-v02.api.letsencrypt.org/your@email.com/keys/your@email.com.key
2018/11/04 20:02:37 [INFO] acme: Registering account for your@email.com
2018/11/04 20:02:38 !!!! HEADS UP !!!!
2018/11/04 20:02:38 
		Your account credentials have been saved in your Let's Encrypt
		configuration directory at "/home/sm/go/src/github.com/xenolf/lego/.lego/accounts/acme-staging-v02.api.letsencrypt.org/your@email.com".
		You should make a secure backup	of this folder now. This
		configuration directory will also contain certificates and
		private keys obtained from Let's Encrypt so making regular
		backups of this folder is ideal.
2018/11/04 20:02:38 [INFO] [mydomain.com, *.mydomain.com] acme: Obtaining bundled SAN certificate
2018/11/04 20:02:38 [INFO] [*.mydomain.com] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz/MAEa-vrkKT6rQuME5GdXeePqH68MwmnZAARUyNxq6GQ
2018/11/04 20:02:38 [INFO] [mydomain.com] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz/WaV77Ah7_R574igB9ZMjh26yiDtJMiOi9RivJL0yJNA
2018/11/04 20:02:38 [INFO] [mydomain.com] acme: Could not find solver for: http-01
2018/11/04 20:02:38 [INFO] [mydomain.com] acme: Preparing to solve DNS-01
2018/11/04 20:02:39 [INFO] [mydomain.com] acme: Preparing to solve DNS-01
2018/11/04 20:02:41 [INFO] [mydomain.com] acme: Trying to solve DNS-01
2018/11/04 20:02:41 [INFO] [mydomain.com] Checking DNS record propagation using [127.0.0.53:53]
2018/11/04 20:02:41 [INFO] Wait [timeout: 4m0s, interval: 2s]
2018/11/04 20:05:21 [INFO] [mydomain.com] The server validated our request
2018/11/04 20:05:21 [INFO] [mydomain.com] acme: Trying to solve DNS-01
2018/11/04 20:05:21 [INFO] [mydomain.com] Checking DNS record propagation using [127.0.0.53:53]
2018/11/04 20:05:21 [INFO] Wait [timeout: 4m0s, interval: 2s]
2018/11/04 20:05:27 [INFO] [mydomain.com] The server validated our request
2018/11/04 20:05:31 [INFO] [mydomain.com, *.mydomain.com] acme: Validations succeeded; requesting certificates
2018/11/04 20:05:32 [INFO] [mydomain.com] Server responded with a certificate.
```

**Comment by ldez** [comment]
Waiting for the merge of https://github.com/smueller18/goinwx/pull/1

**Comment by smueller18** [comment]
merged

**Comment by smueller18** [comment]
Thanks for your review and merging ;-)

**Pull request merged**